Add hoverables (#797)
flame-engine/flame: doc/input.md

**Note**: `HasDraggableComponents` uses an advanced gesture detector under the hood and as explained
further up on this page, shouldn't be used alongside basic detectors.

### Hoverable components

Just like the others, this mixin allows for easy wiring of your component to listen to hover states
and events.

By adding the `HasHoverableComponents` mixin to your base game, and by using the mixin `Hoverable` on

> your components, they get a `isHovered` field and a couple methods (`onHoverStart`, `onHoverEnd`) that

Reviewer: ```suggestion
your components, they get an `isHovered` field and a couple of methods (`onHoverStart`, `onHoverEnd`) that
```

you can override if you want to listen to the events.

```dart

>   bool isHovered = false; // note: you should not modify this value yourself

Reviewer 2: Can't we transform this into a getter, so the user can't change it?

Author: it is a getter, they can always override it right?

Author: on the mixin level I can't control what the impl will do with it

Reviewer 2: I thought it was a field, so da good then

Reviewer 2: Actually, looking on the component, it is a field, I am confused

Author: not sure I understand your suggestions, we need a way to set it and a way to get it, it's a field on the mixin but can be overwritten by the user

Reviewer 2: My suggestion is that this is a getter to the user of the component and internally, we set this using a setter that isn't exposed to the user. That way the user can't change the field at all.

Author: I thought more about it and there might be a way to hide it from the user! (slightly hide it)

  void onHoverEnter(PointerHoverInfo event) {}
  void onHoverLeave(PointerHoverInfo event) {}
```

The provided event info is from the mouse move that triggered the action (entering or leaving).

> While the mouse move keeping inside or outside, no events are fired and those mouse move events are

Reviewer 2: Found this sentence a little confusing, what do you think of this suggestion?

```suggestion
While the mouse movement is kept inside or outside, no events are fired and those mouse move events are
```

> not propagated. Only when the state is changed that the handlers are triggered.

Reviewer: ```suggestion
not propagated. Only when the state is changed the handlers are triggered.
```
